packages/contracts-bedrock/test/cannon/MIPS64Memory.t.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { CommonTest } from "test/setup/CommonTest.sol";
+import { MIPS64Memory } from "src/cannon/libraries/MIPS64Memory.sol";
+import { InvalidMemoryProof } from "src/cannon/libraries/CannonErrors.sol";
+
+contract MIPS64Memory_Test is CommonTest {
+    MIPS64MemoryWithCalldata mem;
+
+    function setUp() public virtual override {
+        super.setUp();
+        mem = new MIPS64MemoryWithCalldata();
+    }
+
+    /// @dev Static unit test for basic memory access
+    function test_readMem_succeeds() external {
+        uint64 addr = 0x100;
+        uint64 word = 0x11_22_33_44_55_66_77_88;
+        bytes32 root;
+        bytes memory proof;
+        (root, proof) = ffi.getCannonMemory64Proof(addr, word);
+        uint64 readWord = mem.readMem(root, addr, 0, proof);
+        assertEq(readWord, word);
+    }
+
+    /// @dev Static unit test asserting that reading from the zero address succeeds
+    function test_readMemAtZero_succeeds() external {
+        uint64 addr = 0x0;
+        uint64 word = 0x11_22_33_44_55_66_77_88;
+        bytes32 root;
+        bytes memory proof;
+        (root, proof) = ffi.getCannonMemory64Proof(addr, word);
+        uint64 readWord = mem.readMem(root, addr, 0, proof);
+        assertEq(readWord, word);
+    }
+
+    /// @dev Static unit test asserting that reading from high memory area succeeds
+    function test_readMemHighMem_succeeds() external {
+        uint64 addr = 0xFF_FF_FF_FF_00_00_00_88;
+        uint64 word = 0x11_22_33_44_55_66_77_88;
+        bytes32 root;
+        bytes memory proof;
+        (root, proof) = ffi.getCannonMemory64Proof(addr, word);
+        uint64 readWord = mem.readMem(root, addr, 0, proof);
+        assertEq(readWord, word);
+    }
+
+    /// @dev Static unit test asserting that reads reverts when a misaligned memory address is provided
+    function test_readInvalidAddress_reverts() external {
+        uint64 addr = 0x100;
+        uint64 word = 0x11_22_33_44_55_66_77_88;
+        bytes32 root;
+        bytes memory proof;
+        (root, proof) = ffi.getCannonMemory64Proof(addr, word);
+        vm.expectRevert(bytes(""));
```

Comment:
Why don't we use revert reasons throughout?  Seems like a reason would be useful here.

Reply:
it's a bit tricky to do this in yul. Lemme find a workaround.

Reply:
Fixed